src/ps2dev.cpp

```diff
@@ -9,307 +9,401 @@
 
 #include "ps2dev.h"
 
-//Enable serial debug mode?
+// Enable serial debug mode?
 //#define _PS2DBG Serial
 
-//since for the device side we are going to be in charge of the clock,
-//the two defines below are how long each _phase_ of the clock cycle is
+// since for the device side we are going to be in charge of the clock,
+// the two defines below are how long each _phase_ of the clock cycle is
 #define CLKFULL 40
+
 // we make changes in the middle of a phase, this how long from the
 // start of phase to the when we drive the data line
 #define CLKHALF 20
 
 // Delay between bytes
-// I've found i need at least 400us to get this working at all,
-// but even more is needed for reliability, so i've put 1000us
+// I've found I need at least 400us to get this working at all,
+// but even more is needed for reliability, so I've put 1000us
 #define BYTEWAIT 1000
 
 // Timeout if computer not sending for 30ms
 #define TIMEOUT 30
 
-
 /*
  * the clock and data pins can be wired directly to the clk and data pins
- * of the PS2 connector.  No external parts are needed.
+ * of the PS2 connector. No external parts are needed.
  */
 PS2dev::PS2dev(int clk, int data)
 {
-  _ps2clk = clk;
-  _ps2data = data;
-  gohi(_ps2clk);
-  gohi(_ps2data);
+	_ps2clk = clk;
+	_ps2data = data;
+	gohi(_ps2clk);
+	gohi(_ps2data);
 }
 
 /*
- * according to some code I saw, these functions will
- * correctly set the clock and data pins for
- * various conditions.  It's done this way so you don't need
- * pullup resistors.
+ * These functions set the state of internal pull-up resistors
  */
-void
-PS2dev::gohi(int pin)
+void PS2dev::gohi(int pin)
 {
-  pinMode(pin, INPUT);
-  digitalWrite(pin, HIGH);
+	DDRB &= ~(1 << pin); // pinMode(pin, INPUT);
+	PORTB |= (1 << pin); // digitalWrite(pin, HIGH);
```

Comment:
This is AVR specific code, which will break compatibility with other microcontrollers that i aim to support. there is reason for arduino HAL.

Reply:
Sounds good.